packages/app-editors/kakoune/kakoune.exlib

```diff
@@ -21,47 +19,47 @@ DEPENDENCIES="
         ]]
 "
 
-WORK=${WORK}/src
-
 pkg_pretend() {
     if cxx-is-gcc && ! ever at_least 7 $(gcc-version); then
         die "${CATEGORY}/${PN} must be built with GCC[>=7] or clang."
     fi
 }
 
+kak_make() {
+    local debug=$(option debug yes no)
+
+    emake \
+        CXX=${CXX} \
+        CXXFLAGS-debug-${debug}="" \
```

Comment:
I am confused by this line, this will disable optimization control based of debug option value. Is that expected ?

Reply:
Yes. It is what I called "remove spurious debug/release CFLAGS the user did not request" in the commit message.

It is generally expected that the user controls the `-O` and `-g` flags through their `/etc/paludis/bashrc`, and the package defaults would have clashed with that. An explicit debug option is generally only used to control project specific debug code or similar, e.g. the `-DKAK_DEBUG` in `CPPFLAGS-debug-yes` for kakoune. Luckily they were in separate flags, so I could overwrite them this way.